canconfig.go

```diff
@@ -0,0 +1,42 @@
+package xsens
+
+import (
+	"fmt"
+)
+
+type CANConfig struct {
+	Enable   bool
+	BaudRate CANBaudRate
+}
+
+const (
+	canCfgEnableOffset   = 2
+	canCfgBaudrateOffset = 3
+	canCfgEnableMask     = byte(0b1)     // Masks reserved bit in Enable byte
+	canCfgBaudrateMask   = ^byte(1 << 7) // Masks reserved bit in BaudRate byte
+)
+
+// Marshal returns the wire representation of the CAN configuration.
+func (o *CANConfig) Marshal() ([]byte, error) {
```

Comment:
Even though the existing APIs don't do this, let's use the encoding.BinaryMarshaler interface here: https://pkg.go.dev/encoding#BinaryMarshaler